### ISSUE TITLE
invoke() not completing when context is provided?

### ISSUE THREAD

**Issue by andrewn** [matthewp/robot-hooks#11]
I have a file upload React component with the following behaviour:

```
wait for files -> check files -> upload to API -> show success message -> reset to waiting after timeout
```

I'd also like to call an `onSuccess` callback when the files are uploaded to tell the parent that it's been done. I thought I'd pass this through the state machine's context object to be called when the upload completes.

I'm trying to use [`invoke`](https://thisrobot.life/api/invoke.html) to perform these but when I provide a context object to the machine, it seems to get stuck. The async actions seems to be completing, but the React component doesn't re-render or it doesn't transition to the next state.

Also, I'm struggling with a good way to debug the transitions.

I've created [an example comparing with and without context](https://codesandbox.io/s/react-robot-invoke-not-completing-when-context-is-provided-24o30?file=/src/App.js). 

Any ideas what I'm doing wrong?

**Comment by matthewp** [comment]
@andrewn I have to go to work, ping me later this week if you don't see it fixed before then. Thanks for finding this tough one!

**Comment by andrewn** [comment]
> Are you seeing the same thing I'm seeing? Where it stays in `idle` permanently? I still see this even if I remove the context from the second machine. I think it's related to the useMachine hook.

No, I see it move to "process" and hang there. But the console outputs that the processing finishes.

[I took a screen recording](https://share.getcloudapp.com/d5uve6Yd) of what I see in Firefox 74.0.1 (macOS 10.14.6).

**Comment by matthewp** [comment]
Ok, it could still be the same bug. I'll try a fix and publish to a tag so you can test it in your example.

**Comment by matthewp** [comment]
Published in 0.3.5, sorry for the delay.

**Comment by andrewn** [comment]
Finally managed to test this and it works fine!

I couldn't work out how to get the Code Sandbox example to update the sub-dependency. But downloading and running locally works perfectly.

Thanks again!
